R/msaBayesianGaugeRR.R

```diff
@@ -0,0 +1,2341 @@
+    ggplot2::coord_equal()
+
+  # theme
+  p <- p +
+    jaspGraphs::themeJaspRaw() +
+    jaspGraphs::geom_rangeframe(sides = "bl")
+
+  tempPlot$plotObject <- p
+
+  contourPlot[["plot"]] <- tempPlot
+
+  # table with the posterior means and CrIs for the risks
+  risksTable <- createJaspTable(title = gettext("Producer's (\u03b4) and Consumer's (\u03b2) Risk"))
+  risksTable$position <- 1
+
+  risksTable$addColumnInfo(name = "risks", title = gettext("Risk"),  type = "string")
+  risksTable$addColumnInfo(name = "means", title = gettext("Mean"),  type = "number")
+  risksTable$addColumnInfo(name = "lower", title = gettext("Lower"), type = "number", overtitle = "95% Credible Interval")
+  risksTable$addColumnInfo(name = "upper", title = gettext("Upper"), type = "number", overtitle = "95% Credible Interval")
+
+  fillDat <- .getRisks(contourDf, mu, options)
+
+  if(isTryError(fillDat)) {
+    errorMsg <- paste("Risks could not be computed:", fillDat, "Try adjusting the specification limits.")
+    risksTable$setError(gettext(errorMsg))
+  } else {
+    risksTable$setData(fillDat)
+  }
+  contourPlot[["table"]] <- risksTable
+
+  return()
+}
+
+.plotPrior <- function(jaspResults, options) {
+  if(!is.null(jaspResults[["priorPlot"]])) {
+    return()
+  }
+  priorPlot <- createJaspContainer(title = gettext("Prior Distribution"))
+  priorPlot$position <- 5
+  priorPlot$dependOn(c("rscalePrior", "report"))
+  jaspResults[["priorPlot"]] <- priorPlot
+
+  gPrior <- createJaspPlot(title = gettext("g-prior"), width = 600, height = 320)
+
+  # axis limit
+  xUpper <- extraDistr::qinvchisq(0.75, nu = 1, tau = options$rscalePrior^2)
+  xBreaks <- jaspGraphs::getPrettyAxisBreaks(c(0, xUpper))
+
+  p <- ggplot2::ggplot() +
+    ggplot2::stat_function(fun = extraDistr::dinvchisq,
+                           args = list(nu = 1, tau = options$rscalePrior^2),
+                           linewidth = 1)
+
+  # axes
+  p <- p + ggplot2::scale_y_continuous("Density") +
+    ggplot2::scale_x_continuous("g", breaks = xBreaks, limits = c(0, xUpper))
+
+  # JASP theme
+  p <- p + jaspGraphs::themeJaspRaw() + jaspGraphs::geom_rangeframe()
+  gPrior$plotObject <- p
+
+  priorPlot[["plot"]] <- gPrior
+
+  return()
+}
+
+### posterior plots
+.plotVariancePosteriors <- function(jaspResults, options, parts, operators){
+
+  if(!is.null(jaspResults[["variancePosteriors"]])){
+    return()
+  }
+
+  variancePosteriors <- createJaspContainer(title = gettext("Posterior Distributions"))
+  variancePosteriors$position <- 6
+  variancePosteriors$dependOn(c(.varCompTableDependencies(),
+                                .postPlotDependencies()))
+  jaspResults[["variancePosteriors"]] <- variancePosteriors
+
+  fits <- jaspResults[["distFit"]][["object"]]
+
+  if(isTryError(fits)) {
+    errorMsg <- paste("The", .getDistNames(options$distType),
+                      "distribution could not be fit to the samples. Try selecting another distribution.")
```

Comment:
should use `gettextf`. If options$distType is limited this can also be 3 times the same sentence with the name already in the string so you can use `gettext` instead of `gettextf`.